query/worker.go

```diff
@@ -308,6 +308,28 @@ nexJobLoop:
 	}
 }
 
+func (w *worker) IsSyncCandidate() bool {
+	return w.peer.IsSyncCandidate()
+}
+
+func (w *worker) IsPeerBehindStartHeight(req ReqMessage) bool {
+	return w.peer.IsPeerBehindStartHeight(req)
+}
+
+// IsWorkerEligibleForBlkHdrFetch is the eligibility function used for the BlockHdrWorkManager to determine workers
+// eligible to receive jobs (the job is to fetch headers). If the peer is not a sync candidate or if its last known
+// block height is behind the job query's start height, it returns false. Otherwise, it returns true.
+func IsWorkerEligibleForBlkHdrFetch(r *activeWorker, next *queryJob) bool {
```

Comment:
This shouldn't be needed. The workers should never be given peers that aren't eligible to be synced off of in the first place.

Reply:
The work manager is not exclusively for fetching block headers. So it would receive connected peer messages [for all peers](https://github.com/lightninglabs/neutrino/blob/64b278771ff75da0b30136af7d4ab0ace06d897e/query/workmanager.go#L170-L173). It just happens that in this use case where it is fetching headers we are only interested in full node peers. That is why I included that there. So a worker could not be useful in fetching block headers but it could be eligible to fetch cfheaders or filters for example.

Reply:
judging by the name, the "next" argument should be known to be a block header fetch. If you accept all types of queries here then the function should be renamed and reworked to decide if the peer is eligible for that request *in general*

Reply:
The name of the function is literally, `IsWorkerEligibleForBlkHdrFetch` does it still ring as generic?

Reply:
The thing is difference between the `queryJob` for fetching block headers and that of fetching cfheaders is the Req field in Request.
https://github.com/Chinwendu20/neutrino/blob/64b278771ff75da0b30136af7d4ab0ace06d897e/query/interface.go#L152-L175
which is defined as an interface, its implementations are defined in the `blockmanager.go`. So dealing with its concrete implementations in query package would lead to a circular dependence.

Reply:
Maybe there is a better way to go about it.